Controllers/conversationController.js

```diff
@@ -108,3 +110,36 @@ exports.show = CatchAsyncError(async (req, res, next) => {
     })
   }
 })
+
+exports.createConversation = CatchAsyncError(async (req, res, next) => {
+  // Find searched user
+  const searchedUser = await User.findOne({
+    raw: true,
+    attributes: ['id'],
+    where: {
+      username: req.body.searchedUserUsername
+    }
+  })
+  // Check if conversation already Exists
+  if (req.body.usersId.includes(searchedUser.id)) {
+    res.status(200).json({
+      status: 'success',
+      message: 'Conversation already exists'
+    })
+  } else {
+    const conversation = await Conversation.create({
+    })
+    await UserConversation.create({
+      userId: req.user.id,
+      conversationId: conversation.dataValues.id
+    })
+    await UserConversation.create({
+      userId: searchedUser.id,
+      conversationId: conversation.dataValues.id
+    })
```

Comment:
This should be put into transaction.
https://sequelize.org/docs/v6/other-topics/transactions/